aten/src/ATen/cpu/vec256/vec256_float_neon.h

```diff
@@ -642,6 +642,853 @@ Vec256<float> inline fmadd(const Vec256<float>& a, const Vec256<float>& b, const
+          __asm__ (
+              "vbsl.f32 %[res_mask], %[in_b], %[in_a]\n\t"
+              : [res_mask] "+w" (vec.values.val[0])
+              : [in_b] "w" (b.values.val[0]), [in_a] "w" (a.values.val[0])
+              :);
+          return vec;
+        }
+      case 2:
+        {
+          Vec256<float> vec;
+          static uint32x4_t mask_low = {0xFFFFFFFF, 0xFFFFFFFF, 0x0, 0x0};
+          vec.values.val[0] = (float32x4_t)mask_low;
+          vec.values.val[1] = a.values.val[1];
+          __asm__ (
+              "vbsl.f32 %[res_mask], %[in_b], %[in_a]\n\t"
+              : [res_mask] "+w" (vec.values.val[0])
+              : [in_b] "w" (b.values.val[0]), [in_a] "w" (a.values.val[0])
+              :);
+          return vec;
+        }
+      case 3:
+        {
+          Vec256<float> vec;
+          static uint32x4_t mask_low = {0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF, 0x0};
+          vec.values.val[0] = (float32x4_t)mask_low;
+          vec.values.val[1] = a.values.val[1];
+          __asm__ (
+              "vbsl.f32 %[res_mask], %[in_b], %[in_a]\n\t"
+              : [res_mask] "+w" (vec.values.val[0])
+              : [in_b] "w" (b.values.val[0]), [in_a] "w" (a.values.val[0])
+              :);
+          return vec;
+        }
+      case 4:
+        return Vec256<float>(b.values.val[0], a.values.val[1]);
+      case 5:
+        {
+          Vec256<float> vec;
+          static uint32x4_t mask_high = {0xFFFFFFFF, 0x0, 0x0, 0x0};
+          vec.values.val[0] = b.values.val[0];
+          vec.values.val[1] = (float32x4_t)mask_high;
+          __asm__ (
+              "vbsl.f32 %[res_mask], %[in_b], %[in_a]\n\t"
+              : [res_mask] "+w" (vec.values.val[1])
+              : [in_b] "w" (b.values.val[1]), [in_a] "w" (a.values.val[1])
+              :);
+          return vec;
+        }
+      case 6:
+        {
+          Vec256<float> vec;
+          static uint32x4_t mask_high = {0xFFFFFFFF, 0xFFFFFFFF, 0x0, 0x0};
+          vec.values.val[0] = b.values.val[0];
+          vec.values.val[1] = (float32x4_t)mask_high;
+          __asm__ (
+              "vbsl.f32 %[res_mask], %[in_b], %[in_a]\n\t"
+              : [res_mask] "+w" (vec.values.val[1])
+              : [in_b] "w" (b.values.val[1]), [in_a] "w" (a.values.val[1])
+              :);
+          return vec;
+        }
+      case 7:
+        {
+          Vec256<float> vec;
+          static uint32x4_t mask_high = {0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF, 0x0};
+          vec.values.val[0] = b.values.val[0];
+          vec.values.val[1] = (float32x4_t)mask_high;
+          __asm__ (
+              "vbsl.f32 %[res_mask], %[in_b], %[in_a]\n\t"
+              : [res_mask] "+w" (vec.values.val[1])
+              : [in_b] "w" (b.values.val[1]), [in_a] "w" (a.values.val[1])
+              :);
+          return vec;
+        }
+    }
+    return b;
+  }
+  static Vec256<float> loadu(const void* ptr, int64_t count = size()) {
+    if (count == size()) {
```

Comment:
Even though size() is a constexpr function it can be evaluated at runtime.  constexpr functions are only required to be evaluated at compile-time if they are used in a context that requires compile-time computation, for instance in a static_assert or as size of an array.  If you want guarantee that this will be evaluated at compile-time (since you are calling this function several times) then you have to assign it to constexpr variable.  That's all in theory though - in practice you should be fine.

Reply:
But `count` is not constexpr variable. I am relying on size to be conexpred but besides that I cannot do anything to ensure size() is guaranteed to constexpr. At least thats what I understand.